price-feeder/oracle/provider/binance.go

```diff
@@ -270,6 +266,16 @@ func (p *BinanceProvider) messageReceived(messageType int, bz []byte) {
 			"provider",
 			config.ProviderBinance,
 		)
+		return
+	}
+
+	if tickerErr != nil || candleErr != nil {
+		p.logger.Error().
+			Int("bz_length", len(bz)).
+			Str("bz", string(bz[:maxSizeMessageInLog])).
```

Comment:
imo this is a bit messy, is there a way to set a max message length on the logger class itself? Maybe we just don't print the bz at all if there's not a cleaner way

Reply:
The zerolog doesn't have an option to limit the number of characters inside the logger 
We could maybe instead of printing in string we could use bytes `Bytes("bz", bz[:maxSizeMessageInLog]).` but probably it would keep a bit messy 

If we remove the message sent by the provider when logging errors, it could make it more difficult to debug and get the edge cases 

What do you guys think? @toteki @adamewozniak

Reply:
A couple things:
- we still need to remove that `if` statement. If we reach the end of the function (successes would have returned early), there must be a log created.
- the two cases are (a) both errors non-nil and (b) edge case where an err was nil, but a critical field was zero

I guess if we log the errors, we can detect the edge case just by the fact that one was `<nil>`, so that can be ignored.

Perhaps replace this
```
if tickerErr != nil || candleErr != nil {
	p.logger.Error().
		Int("bz_length", len(bz)).
		Str("bz", string(bz[:maxSizeMessageInLog])).
		AnErr("ticker", tickerErr).
		AnErr("candle", candleErr).
		Msg("Error on receive message")
}
```

with this:
```
e := p.logger.Error().
	Int("length", len(bz)).
	AnErr("ticker", tickerErr).
	AnErr("candle", candleErr).
	Msg("Error on receive message")
```

At least the message length can be logged in a neat way.

Still it might be nice to know what a message looked like if both unmarshals failed - might not be a neat way of doing that though.

Reply:
done at https://github.com/umee-network/umee/pull/684/commits/d6a500694c3c01e4cf0357a92b787a9b541daa9f